part2/bowling/turn.rb

```diff
@@ -0,0 +1,33 @@
+class Turn
+  attr_accessor :frames
+
+  def initialize
+    @frames = [Frame.new]
+  end
+
+  def roll(pins)
+    frames.last << pins
+    frames[0...-1].select(&:extra_points?).each do |frame|
+      frame.extra << pins
+    end
+    if frames.count != 10 && frames.last.done?
+      frames << Frame.new
+    end
+  end
```

Comment:
Consider refactoring this out into a couple of methods - each one should be in charge of doing only one thing.